packages/dds/tree/src/tableSchema.ts

```diff
@@ -1322,12 +1323,20 @@ export namespace TableSchema {
 		/**
 		 * The table's columns.
 		 */
-		readonly columns: TreeArrayNode<TColumn>;
+		readonly columns: ReadonlyArrayNode<
```

Comment:
@CraigMacomber seems like we should probably have a type like `TreeArrayNode` that does this type computation for us. `ReadonlyArrayNode`'s type arguments are different (simpler). Would it make sense for me to add a type for this?

Reply:
Is there a reason not to just do `TreeNodeFromImplicitAllowedTypes<TColumn>` here? I'm unclear on what the extra extends logic is supposed to accomplish. Both TreeNodeFromImplicitAllowedTypes and ReadonlyArrayNode should behave covariantly, so extra care or tricks should not be needed here.

Perhaps this is some trick to make this work better in the generic context?

Reply:
Ah, I forgot this was equivalent to `TreeNodeFromImplicitAllowedTypes<TColumn>`. That's easy enough to update then. I was initially thrown by `TreeArrayNode` taking our `TColumn` directly and `ReadonlyArrayNode` expecting something else.